src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -570,17 +665,26 @@ PosixEventEngine::GetDNSResolver(
   // configuration.
   if (ShouldUseAresDnsResolver()) {
 #if GRPC_ARES == 1 && defined(GRPC_POSIX_SOCKET_ARES_EV_DRIVER)
+#if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
     GRPC_TRACE_LOG(event_engine_dns, INFO)
         << "PosixEventEngine::" << this << " creating AresResolver";
     auto ares_resolver = AresResolver::CreateAresResolver(
         options.dns_server,
-        std::make_unique<GrpcPolledFdFactoryPosix>(poller_manager_->Poller()),
+        std::make_unique<GrpcPolledFdFactoryPosix>(poller_manager_.Poller()),
         shared_from_this());
     if (!ares_resolver.ok()) {
       return ares_resolver.status();
     }
+#if GRPC_ENABLE_FORK_SUPPORT && GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
+    grpc_core::MutexLock lock(&mu_);
```

Comment:
I assume there should be braces around these two statements, so that we're not still holding the lock when we create the DNS resolver to return?